extensions/wikia/FounderEmails/FounderEmails.php

```diff
@@ -53,33 +48,36 @@
  */
 $wgExtensionFunctions[] = 'wfFounderEmailsInit';
 
-function wfFounderEmailsInit() {
-	global $wgHooks, $wgAutoloadClasses, $wgFounderEmailsEvents, $wgDefaultUserOptions, $wgCityId;
+function wfFounderEmailsInit()
+{
+	global $wgHooks, $wgAutoloadClasses;
 
-	$dir = dirname( __FILE__ ) . '/';
+	$dir = dirname(__FILE__) . '/';
 
 	/**
 	 * classes
 	 */
 	$wgAutoloadClasses['FounderEmails'] = $dir . 'FounderEmails.class.php';
 	$wgAutoloadClasses['FounderEmailsEvent'] = $dir . 'FounderEmailsEvent.class.php';
+	$wgAutoloadClasses['FounderEmailsEditEvent'] = $dir . 'events/FounderEmailsEditEvent.class.php';
+	$wgAutoloadClasses['FounderEmailsRegisterEvent'] = $dir . 'events/FounderEmailsRegisterEvent.class.php';
+	$wgAutoloadClasses['FounderEmailsDaysPassedEvent'] = $dir . 'events/FounderEmailsDaysPassedEvent.class.php';
+	$wgAutoloadClasses['FounderEmailsCompleteDigestEvent'] = $dir . 'events/FounderEmailsCompleteDigestEvent.class.php';
+	$wgAutoloadClasses['FounderEmailsViewsDigestEvent'] = $dir . 'events/FounderEmailsViewsDigestEvent.class.php';
 
-	// add event classes & hooks
-	foreach ( $wgFounderEmailsEvents as $event ) {
-		$wgAutoloadClasses[$event['className']] = $dir . 'events/' . $event['className'] . '.class.php';
-		if ( !empty( $event['hookName'] ) ) {
-			$wgHooks[$event['hookName']][] = $event['className'] . '::register';
-		}
-	}
+	$wgHooks['RecentChange_save'][] = 'FounderEmailsEditEvent::recentChanges';
+	$wgHooks['AddNewAccount'][] = 'FounderEmailsRegisterEvent::registerUser';
+	$wgHooks['CreateWikiLocalJob-complete'][] = 'FounderEmailsDaysPassedEvent::createWikiCompleted';
 
 	$wgHooks['GetPreferences'][] = 'FounderEmails::onGetPreferences';
 	$wgHooks['UserRights'][] = 'FounderEmails::onUserRightsChange';
 }
 
-$dir = dirname( __FILE__ ) . '/';
+$dir = dirname(__FILE__) . '/';
 $wgAutoloadClasses['FounderEmailsController'] = $dir . 'FounderEmailsController.class.php';
 $wgAutoloadClasses['SpecialFounderEmails'] = $dir . 'SpecialFounderEmails.class.php';
 
+global $wgCityId;
```

Comment:
`global` not needed - setup files are evaluated in global scope

Reply:
it's still good practice to use explicit scoping rather than rely on implicit global scope

Reply:
Then we should add `global $wgAutoloadClasses` as well ;)